### ISSUE TITLE
'returns void' causing proxy generation errors.

### ISSUE THREAD

**Issue by IceReaper** [doctrine/doctrine2#6000]
When using a setter which returns nothing / void:

    /** @Column(type="string", unique=true) */
    protected /*string*/ $name;
    public function getName() : string { return $this->name; }
    public function setName(string $name) : void { $this->name = $name; }

Doctrine generates this:

    /**
     * {@inheritDoc}
     */
    public function setName(string $name): void
    {

        $this->__initializer__ && $this->__initializer__->__invoke($this, 'setName', [$name]);

        return parent::setName($name);
    }

The error message is:

    PHP Fatal error:  A void function must not return a value in <...> on line <...>

**Comment by Ocramius** [comment]
We are not yet supporting PHP 7.1, since the reflection API hasn't
stabilised yet.

**Comment by IceReaper** [comment]
Oh, sorry, didnt expect the reflection API to change after reaching RC. However, might be interesting to keep this bug open, as its very likely "void" wont change, and this bug will still affect.

**Comment by IceReaper** [comment]
As a workaround till official support lands, i changed the last block in class ProxyGenerator in method generateMethods in line 797, just in case someone comes across the same problem:

    $return = strval($method->getReturnType()) === 'void' ? '' : 'return';

    $methods .= "\n        \$this->__initializer__ "
        . "&& \$this->__initializer__->__invoke(\$this, " . var_export($name, true)
        . ", [" . $invokeParamsString . "]);"
        . "\n\n        " . $return . " parent::" . $name . '(' . $callParamsString . ');'
        . "\n" . '    }' . "\n";

**Comment by Majkl578** [comment]
See doctrine/common#734. Void is another thing that needs to be done

**Comment by Ocramius** [comment]
In general, since the reflection API in 7.1 is quite messy on `ReflectionType`, avoid putting any work in this until 7.1.0 is stable. If you still want to do that, then please ensure full backwards compatibility without switch/decision paths (no 7.1.0 API level checking, please, unless for new features)

**Comment by Majkl578** [comment]
This should be now fixed by doctrine/common#734 which has been merged just now.
If you don't want to wait for next doctrine/orm release with the patch included, you can easily try it out e.g. by running:
```bash
composer require doctrine/common 'dev-master#5086700 as 2.6.99'
```

**Comment by Ocramius** [comment]
Indeed, this will require `doctrine/common` `2.7.0`
